lane.py

```diff
@@ -638,6 +662,58 @@ def quality_tier_field(self):
 
 
 class SearchFacets(FacetsWithEntryPoint):
+    """A Facets object designed to filter search results.
+
+    Most search result filtering is handled by WorkList, but this
+    allows someone to, e.g., search a multi-lingual WorkList in their
+    preferred language.
+    """
+    
+    def __init__(self, entrypoint=None, media=None, languages=None, **kwargs):
+        super(SearchFacets, self).__init__(entrypoint, **kwargs)
+        if media == Edition.ALL_MEDIUM:
+            self.media = media
+        else:
+            self.media = self._ensure_list(media)
+
+        self.languages = self._ensure_list(languages)
+
+    def _ensure_list(self, x):
+        """Make sure x is a list of values, if there is a value at all."""
+        if x is None:
+            return None
+        if isinstance(x, list):
+            return x
+        return [x]
+
+    @classmethod
+    def from_request(cls, library, config, get_argument, get_header, worklist,
+                     **extra):
+
+        # Searches against a WorkList that has no particular language
+        # restrictions will use the languages defined in the
+        # Accept-Language header used by the client.
+        language_header = get_header("Accept-Language")
+        languages = None
+        if language_header:
+            languages = parse_header(language_header)
+            languages = map(str, languages)
+            languages = map(LanguageCodes.iso_639_2_for_locale, languages)
+            languages = [l for l in languages if l]
+        languages = languages or None
```

Comment:
This code is copied in from circulation; my circulation branch will remove it.

Reply:
I know this was existing functionality, but while we're here, this is a problem in the admin interface. I've gotten complaints that admins can't search for non-english books.

Reply:
I filed https://jira.nypl.org/browse/SIMPLY-1213 to deal with this.